pandas/core/generic.py

```diff
@@ -4241,9 +4241,9 @@ def reindex(self: FrameOrSeries, *args, **kwargs) -> FrameOrSeries:
         """
         Conform %(klass)s to new index with optional filling logic.
 
-        Places NA/NaN in locations having no value in the previous index. A new object
-        is produced unless the new index is equivalent to the current one and
-        ``copy=False``.
+        Places NA/NaN in locations having no value in the previous index. A new
+        DataFrame with an additional index is generated subject to customization
```

Comment:
No `reindex` does not generate additional index for `DataFrame` or `Series`. "A new object" actually refers to a new `DataFrame` or `Series`, not a new index. Perhaps you can try the function to see it's behavior and then decide your wordings